### ISSUE TITLE
Unhandled error event

### ISSUE THREAD

**Issue by csulok** [vesse/passport-ldapauth#29]
If there's a DNS related error when creating the LDAP client, I get an unhandled error event. From the description given by the ldapjs maintainers, it looks like whatever is creating the client should have the error event listener. As I see it, the client instance is not exposed through passport, so I wonder if anything can be done in passport-ldapauth.

https://github.com/mcavage/node-ldapjs/issues/234

Currently I have mcavage/node-ldapjs#773a1c6 and passport-ldapauth ^0.3.0 installed.

``` js
events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: getaddrinfo ENOTFOUND example.asd
    at errnoException (dns.js:44:10)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:94:26)
```

**Comment by vesse** [comment]
I need to check this when I have time. The  `ldapjs` clients are in [vesse/node-ldapauth-fork] which this passport wrapper uses, and they do not have the listener attached. It's easy to add the listeners, but I don't know yet at which point this error really happens and thus what could be done when it happens.

**Comment by csulok** [comment]
The stacktrace provided by Node is awful from this point of view, no way to easily find out where the request is initiated in any of 3-4 modules. Anyhow your comment is helpful, I will investigate further in the other modules and open issues there as appropriate.

**Comment by vesse** [comment]
Should work now with `ldapjs 1.0.0`, see [test case](https://github.com/vesse/passport-ldapauth/blob/0b7e1d0c4b065e979ad25996417b5ee8a68e7ad4/test/strategy-test.js#L328-L343)

**Comment by chriswininger** [comment]
also bumping up against this

**Comment by chriswininger** [comment]
Seems like the heart of this is with ldapauth-fork, https://github.com/vesse/node-ldapauth-fork/pull/41, if they accept that pull request then it should be trivial to fix the problem in passport-lpapauth

**Comment by chriswininger** [comment]
Here's a version of passport-ldapauth which illustrates how by pulling in the the fix for node-ldapauth-fork we could patch passport-ldapauth. I didn't make this an official pull request because the pull request for node-ldapauth-fork#41 hasn't been accepted yet, but if/when that gets accepted and merged into the main ldapauth repo you should be able to do something like this

https://github.com/chriswininger/passport-ldapauth/commit/9064385e62352ac96ea16cee1264428932bf3e0c

It works as a temporary work around for us anyways

**Comment by vesse** [comment]
I believe this should now be fixed with `1.0.0` of `passport-ldapauth` since `ldapauth-fork` now re-emits `ldapjs` errors.

**Comment by chriswininger** [comment]
That's great! Thanks
